cpp/open3d/utility/Parallel.cpp

```diff
@@ -7,51 +7,21 @@
 
 #include "open3d/utility/Parallel.h"
 
-#ifdef _OPENMP
-#include <omp.h>
-#endif
-
-#include <cstdlib>
-#include <string>
-
-#include "open3d/utility/CPUInfo.h"
-#include "open3d/utility/Logging.h"
+#include <tbb/task_arena.h>
 
 namespace open3d {
 namespace utility {
 
-static std::string GetEnvVar(const std::string& name) {
-    if (const char* value = std::getenv(name.c_str())) {
-        return std::string(value);
-    } else {
-        return "";
-    }
-}
+int EstimateMaxThreads() { return tbb::this_task_arena::max_concurrency(); }
 
-int EstimateMaxThreads() {
-#ifdef _OPENMP
-    if (!GetEnvVar("OMP_NUM_THREADS").empty() ||
-        !GetEnvVar("OMP_DYNAMIC").empty()) {
-        // See the full list of OpenMP environment variables at:
-        // https://www.openmp.org/spec-html/5.0/openmpch6.html
-        return omp_get_max_threads();
-    } else {
-        // Returns the number of physical cores.
-        return utility::CPUInfo::GetInstance().NumCores();
-    }
-#else
-    (void)&GetEnvVar;  // Avoids compiler warning.
-    return 1;
-#endif
+std::size_t& DefaultGrainSizeTBB() noexcept {
+    static std::size_t GrainSize = 256;
```

Comment:
Honestly, I was guessing at grain size [from this](https://oneapi-src.github.io/oneTBB/main/tbb_userguide/Controlling_Chunking_os.html), but it really should be picked based off of profiling. My understanding is that the grain size provides loose guidance to TBB's automatic chunking mechanism. It works similarly to `omp schedule(guided)`. Overall the goal is to provide plenty of work to each thread so the overhead of chunking is minimized, but small enough chunks that the scheduler can go back in a steal some if one of the threads gets held up. It might be worth taking another pass through the grain sizes that I put in and set them as a magic number times the `DefaultGrainSizeTBB` (which is mutable). That way the chunk size could be higher for doing a single operation with tensors, and smaller when looping through complex sections like in RANSAC.